### PR TITLE
workload: rate limit splits to prevent replica imbalance

### PR DESCRIPTION
**Pull request by ajwerner**
Before this change when applying a large number of splits to a cluster the
splitting would occur faster than the empty replicas can be moved to unloaded
nodes. This leads to massive imbalance between replicas at the end of the
splitting process. The value of 128 was selected experimentally, at 256
divergence was observed. Waiting for the cluster to rebalance ends up being
slower than preventing the imbalance in the first place. 

**Before**
before

**After**
after

Release note: None

### CODE REVIEW

File: pkg/workload/workload.go [hunk 1]
```diff
@@ -435,6 +436,8 @@ func Split(ctx context.Context, db *gosql.DB, table Table, concurrency int) erro
 
 	log.Infof(ctx, `starting %d splits`, len(splitPoints))
 	g := ctxgroup.WithContext(ctx)
+	// Rate limit splitting to prevent replica imbalance.
+	r := rate.NewLimiter(128, 64)
```
Comment (nvanbenschoten) [inline pkg/workload/workload.go]: What's the point of specifying a burst size?

**Review by nvb — COMMENTED** [review]
Nice, this is a good idea.

File: pkg/workload/workload.go (hunk 1 above)
Comment (ajwerner) [reply pkg/workload/workload.go]: My original idea was to not slow down small numbers of splits. I had initially also picked a much lower rate. Mostly just picked a number without much thinking and didn't revisit it. It's now 0.

Comment (ajwerner) [reply pkg/workload/workload.go]: Burst needs to be at least 1.

**Review by nvb — APPROVED** [review]

**Pull request merged**